internal/webhook/external/azure.go

```diff
@@ -0,0 +1,188 @@
+package external
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/git"
+	"github.com/akuity/kargo/internal/helm"
+	xhttp "github.com/akuity/kargo/internal/http"
+	"github.com/akuity/kargo/internal/image"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+const (
+	imageMediaType     = "application/vnd.docker.distribution.manifest.v2+json"
+	azureSecretDataKey = "secret"
+	azure              = "azure"
+)
+
+func init() {
+	registry.register(
+		azure,
+		webhookReceiverRegistration{
+			predicate: func(cfg kargoapi.WebhookReceiverConfig) bool {
+				return cfg.Azure != nil
+			},
+			factory: newAzureWebhookReceiver,
+		},
+	)
+}
+
+// azureWebhookReceiver is an implementation of WebhookReceiver that handles
+// inbound webhooks from both Azure Container Registry (ACR) and Azure DevOps.
+type azureWebhookReceiver struct {
+	*baseWebhookReceiver
+}
+
+// newAzureWebhookReceiver returns a new instance of azureWebhookReceiver.
+func newAzureWebhookReceiver(
+	c client.Client,
+	project string,
+	cfg kargoapi.WebhookReceiverConfig,
+) WebhookReceiver {
+	return &azureWebhookReceiver{
+		baseWebhookReceiver: &baseWebhookReceiver{
+			client:     c,
+			project:    project,
+			secretName: cfg.Azure.SecretRef.Name,
+		},
+	}
+}
+
+// getReceiverType implements WebhookReceiver.
+func (q *azureWebhookReceiver) getReceiverType() string {
```

Comment:
Guessing the Quay receiver was used as a sort of template. This will have to change in a few spots.

```suggestion
func (a *azureWebhookReceiver) getReceiverType() string {
```